### PR TITLE
Updated for 9 class product

### PR DESCRIPTION
**Pull request by SeffatChowdhury**
@alexgleith please review

### CODE REVIEW

**Review by alexgleith — CHANGES_REQUESTED** [review]
I don't think there's any need to remove the classes that aren't used, and you need to change the name of the product, so that it matches the name on the PC.

I think actually duplicate this document and just change the product name inside and as the file name.

File: products/io_lulc.odc-product.yaml
Comment (alexgleith) [inline products/io_lulc.odc-product.yaml:31]: Please remove this

**Review by alexgleith — CHANGES_REQUESTED** [review]
Something s there still.

Has this been tested?

File: products/io_lulc.odc-product.yaml
Comment (SeffatChowdhury) [reply products/io_lulc.odc-product.yaml:31]: The line or the line break difference?

Comment (SeffatChowdhury) [reply products/io_lulc.odc-product.yaml:31]: What's the best way to test it without creating a dummy product in the DEV env?

Comment (SeffatChowdhury) [reply products/io_lulc.odc-product.yaml:31]: @alexgleith

**Review by alexgleith — APPROVED** [review]

**Review by fangfy — COMMENTED** [review]
Is the only differencece between "io_lulc" and "io_lulc_9_class" the use of class 11? if that's the case I suggest just updating existing one with "11: rangeland". It will still work for the old version.

**Comment by SeffatChowdhury** [comment]
@fangfy Alex suggested to add the 9 class product def separately. Two differences: class 11 is used and two other classes are not used (Alex suggested to still keep them)

**Comment by alexgleith** [comment]
The way that `stac-to-dc` tool works, it takes the name from the Microsoft Planetary Computer STAC API Collection, so you need two products anyway.

There's no downside to having two product definitions, one for each of the products listed on the STAC API.

Regarding testing, you can create a local ODC instance and do it, or do it in the DE Africa DEV cluster, that's totally up to you!

**Comment by fangfy** [comment]
> @fangfy Alex suggested to add the 9 class product def separately. Two differences: class 11 is used and two other classes are not used (Alex suggested to still keep them)

why are we keeping them if they are not used? if we keep unused classes then the two can be made exactly the same (except product name)

**Comment by fangfy** [comment]
> 



> The way that `stac-to-dc` tool works, it takes the name from the Microsoft Planetary Computer STAC API Collection, so you need two products anyway.
> 
> There's no downside to having two product definitions, one for each of the products listed on the STAC API.
> 
> Regarding testing, you can create a local ODC instance and do it, or do it in the DE Africa DEV cluster, that's totally up to you!

OK.... so PC is calling this second product io-lulc-9-class, which is a terrible name IMHO. Do we have to keep this the same as PC?  I just want us to have one io_lulc product in deafrica

**Comment by alexgleith** [comment]
No, it doesn't need to be named the same, but... you will need to change the indexing script to handle renaming it, which is a good idea.

This needs a new parameter added to it is all: https://github.com/opendatacube/odc-tools/blob/develop/apps/dc_tools/odc/apps/dc_tools/stac_api_to_dc.py

**Comment by SeffatChowdhury** [comment]
@alexgleith could you advise on how to contribute to repo doc-tools?

**Comment by fangfy** [comment]
> No, it doesn't need to be named the same, but... you will need to change the indexing script to handle renaming it, which is a good idea.
> 
> This needs a new parameter added to it is all: https://github.com/opendatacube/odc-tools/blob/develop/apps/dc_tools/odc/apps/dc_tools/stac_api_to_dc.py

@SeffatChowdhury let me know if you'd like to try update this tool. I can review but I think you'll need to set up to be able to test the changes locally. 

Regardless of the name though, the "flags_definition" need to match the actual product, so 9 labels listed for the 9 class lulc map.

**Comment by SeffatChowdhury** [comment]
@fangfy please assign me write access to odc-tools if possible, I can update it and will be able to test locally in Sandbox

**Comment by fangfy** [comment]
> 

> @fangfy please assign me write access to odc-tools if possible, I can update it and will be able to test locally in Sandbox

Change to this script will impact others, so you'll have to test it in a local odc installation before you can push and merge in the changes.

**Comment by fangfy** [comment]
> 

> @fangfy please assign me write access to odc-tools if possible, I can update it and will be able to test locally in Sandbox

Change to this script will impact others, so you'll have to test it in a local odc installation before you can push and merge in the changes.

**Comment by SeffatChowdhury** [comment]
@fangfy that's ok, I do have a local ODC installation (non DEA) for testing

**Comment by alexgleith** [comment]
Hey @SeffatChowdhury you can create a pull request without needing write access to the repo. I'm happy to review the PR.

**Comment by SeffatChowdhury** [comment]
@fangfy please join this repo and review the PR https://github.com/opendatacube/odc-tools/pull/493 Thanks

**Comment by SeffatChowdhury** [comment]
@fangfy I'll revert to "not used", but as per @alexgleith review earlier, the list does need to be exhaustive

**Pull request merged**
